stable/concourse/templates/web-deployment.yaml

```diff
@@ -30,62 +30,51 @@ spec:
-            - "--aws-ssm-region"
-            - "$(CONCOURSE_AWS_SSM_REGION)"
+            {{- if .Values.concourse.web.bindIp }}
+            - name: CONCOURSE_BIND_IP
+              value: {{ .Values.concourse.web.bindIp | quote }}
             {{- end }}
-            {{- if .Values.credentialManager.awsSecretsManager.enabled }}
-            - "--aws-secretsmanager-region"
-            - "$(CONCOURSE_AWS_SECRETSMANAGER_REGION)"
+            {{- if .Values.concourse.web.localAuth.enabled }}
+            - name: CONCOURSE_ADD_LOCAL_USER
+              value: {{ .Values.secrets.localUsers }}
             {{- end }}
-            {{- if .Values.web.metrics.datadog.enabled }}
-            - "--datadog-agent-host"
-            - "$(CONCOURSE_DATADOG_AGENT_HOST)"
-            - "--datadog-agent-port"
-            - "$(CONCOURSE_DATADOG_AGENT_PORT)"
-            {{- if .Values.web.metrics.datadog.prefix }}
-            - "--datadog-prefix"
-            - {{ .Values.web.metrics.datadog.prefix }}
+            {{- if .Values.concourse.web.tlsBindPort }}
+            - name: CONCOURSE_TLS_BIND_PORT
+              value: {{ .Values.concourse.web.tlsBindPort | quote }}
             {{- end }}
+            {{- if .Values.concourse.web.tlsCert }}
+            - name: CONCOURSE_TLS_CERT
+              value: {{ .Values.concourse.web.tlsCert }}
             {{- end }}
-          env:
-            - name: CONCOURSE_TSA_HOST_KEY
-              value: "/concourse-keys/host_key"
-            - name: CONCOURSE_SESSION_SIGNING_KEY
-              value: "/concourse-keys/session_signing_key"
-            - name: CONCOURSE_TSA_AUTHORIZED_KEYS
-              value: "/concourse-keys/worker_key.pub"
-            {{- if .Values.postgresql.enabled }}
-            - name: POSTGRES_HOST
-              value: {{ template "concourse.postgresql.fullname" . }}
-            - name: POSTGRES_USER
-              value: {{ .Values.postgresql.postgresUser }}
-            - name: POSTGRES_PASSWORD
-              valueFrom:
-                secretKeyRef:
-                  name: {{ template "concourse.postgresql.fullname" . }}
-                  key: postgres-password
-            - name: POSTGRES_DATABASE
-              value: {{ .Values.postgresql.postgresDatabase | quote }}
-            - name: CONCOURSE_POSTGRES_DATA_SOURCE
-              value: postgres://$(POSTGRES_USER):$(POSTGRES_PASSWORD)@$(POSTGRES_HOST)/$(POSTGRES_DATABASE)?sslmode=disable
+            {{- if .Values.concourse.web.tlsKey }}
+            - name: CONCOURSE_TLS_KEY
+              value: {{ .Values.concourse.web.tlsKey }}
+            {{- end }}
+            {{- if .Values.concourse.web.externalUrl }}
+            - name: CONCOURSE_EXTERNAL_URL
+              value: {{ .Values.concourse.externalURL | quote }}
+            {{- end }}
+            {{- if .Values.concourse.web.peerUrl }}
+            - name: CONCOURSE_PEER_URL
+              value: {{ .Values.concourse.web.peerUrl }}
```

Comment:
When would we want this to not be the pod IP? It would have to be configured on a per ATC basis, which this chart isn't capable of as it stands

Reply:
Not sure here as we don't know all edge cases of k8s deployment so we have the if there for flexibility. If there is no such case it could be removed so peer URL will be always pod IP